PERSONAL_DATA_DISCLOSURE.md

```diff
@@ -0,0 +1,87 @@
+_Last Updated: Mon, 28 Nov 2022 UTC_
+_Last Reviewed: Mon, 28 Nov 2022 UTC_
+
+## Overview
+
+The purpose of this form is to disclose the types of personal data[^1] (PD) stored by each module.  This information enables those hosting FOLIO to better manage and comply with various privacy laws and restrictions, e.g. GDPR.
+
+It's important to note that PD is not limited to that which can be used to identify a person on it's own (e.g. Social security number), but also data used in conjunction with other data to identify a person (e.g. date of birth + city + gender, or dynamic IP address + date + time), and any information about such a person (e.g. has loaned 5 items, 2 of which are overdue).
+
+For the purposes of this form, "store" includes the following:
+
+* Persisting to storage - Either internal (e.g. Postgres) or external (e.g. S3, etc.) to FOLIO
+* Caching - In-memory, etc.
+* Logging
+* Sending to an external piece of infrastructure such as a queue (e.g. Kafka), database (e.g. Elasticsearch, Library Data Platform), distributed table, etc.
+
+## Personal Data processed by this Module
+
+- [ ] This module does not store any PD.
```

Comment:
I know that `storage` is an ambiguous term in this form and that makes it hard to be confident answers this question.

As I understand it, mod-users-bl does not have any persistent storage and so I would be comfortable stating no storage. Have I missed something?

Reply:
This is going to/from the database as per
> Sending to an external piece of infrastructure such as a queue (e.g. Kafka), database (e.g. Elasticsearch, Library Data Platform), distributed table, etc.

However, you might be able to argue that the database it is going to is "internal" rather than "external".

I would also double-check whether or not any of the user information ends of in any Kafka messages.
If any part of the relevant user data ends up in a Kafka message then that is considered "stores" by this form.

Reply:
> However, you might be able to argue that the database it is going to is "internal" rather than "external".

I'm not sure I follow this. Which database are you referring to?

As I understand it the storage / business logic split between mod-users and mod-users-bl means that mod-users is the module with storage responsibility and mod-users-bl does not.

Are you suggesting that mod-users-bl should be considered as storing something because it uses mod-users?

I'm asking because if we apply that level of transitivity, pretty much all modules would be considered storing.

Reply:
The module does store password changes-which is why I did not mark this.

Reply:
Oh, I did not expect that.

Can you refer me to the code where it stores those password changes?

Reply:
Hmm, on closer inspection, it seems to be using the /authn/update endpoint-I somehow got the impression it was communicating with the back-end database.  So it's using another module endpoint, and is not itself storing user credentials.

Reply:
Cool, thanks for digging further